.github/workflows/maven-check-deps-build-publish.yml

```diff
@@ -122,96 +122,60 @@ jobs:
-          
-          echo "List of modules to check for dependency changes:"
-          echo "$modules"
-          
-          # Build the local dependency report
-          echo "Compile a local dependency report..."
-          mvn --batch-mode --no-transfer-progress clean $MAVEN_PHASE $MAVEN_ARGS
-          
-          # Iterate over each module
-          for module in $modules; do
-            modulePath=$(echo $module | sed 's/\\/\//g')
-          
-            # Navigate to the module directory
-            cd $(dirname $modulePath)
-          
-            # Get the groupId, artifactId, and version
-            groupId=$(mvn help:evaluate -Dexpression=project.groupId -q -DforceStdout)
-            artifactId=$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)
-            version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
-            classifier=$(mvn help:evaluate -Dexpression=dependencyReportClassifier -q -DforceStdout)
-            artifact="${groupId}":"${artifactId}":"${version}":txt:"${classifier}"
-          
-            filename=${artifactId}-${version}-${classifier}.txt
-            absolutePath=/tmp/$filename
-          
-            # Remove previous file if any
-            rm -f "${absolutePath}"
-          
-            # Fetch the remote dependency report
-            echo "Fetch remote dependency report ${artifact}..."
-            set +e
-            mvn --batch-mode --no-transfer-progress org.apache.maven.plugins:maven-dependency-plugin:3.6.0:get -DremoteRepositories=${remoteRepoUrl} -Dartifact=${artifact} -Dtransitive=false
-            mvn --batch-mode --no-transfer-progress org.apache.maven.plugins:maven-dependency-plugin:3.6.0:copy -Dartifact=${artifact} -DoutputDirectory=/tmp/ -Dmdep.useBaseVersion=true
-            set -e
-          
-            # If no dependency report was fetched, create an empty one.
-            if [ ! -f "$absolutePath" ]; then
-              echo "Remote dependency file is not found at '${absolutePath}'. Creating an empty one and continue."
-              touch "${absolutePath}"
-            else
-              echo "Remote dependency file saved at '${absolutePath}'."
-            fi
-          
-            # Compare the 2 files. Will exit with 0 if no change, 1 if changes
-            echo "Compare both dependency reports..."
-            set +e
-            if [ -f ./target/"${filename}" ]; then
-              diff "${absolutePath}" ./target/"${filename}"
-              diff_rc=$?
-            
-            if [ $diff_rc -eq 0 ]; then
-              echo "No dependency change. Exit (0)"
-            elif [ $diff_rc -eq 1 ]; then
-              echo "One or more dependency has changed. Exit (1)"
-               # Set hasChanged to 1 (true) if a difference is found
+
+          # Run the build (comparison will be done during build)
+          mvn --batch-mode --no-transfer-progress clean compile
```

Comment:
How does [dependency-tracker-maven-plugin](https://github.com/mekomsolutions/dependency-tracker-maven-plugin) handle multi module projects? How many `*-comparison.txt` does it produce? one per submodule?

Reply:
It's one per module.
So the bash logic should search for those files across all modules and re-publish if any is not 0.

Reply:
I've added logic to search for any dependency reports in the project.